components/form/validate.ts

```diff
@@ -40,21 +34,27 @@ export const resolveValue = (object: any, path: string) =>
   parsePath(path)
     .reduce((o, p) => o ? o[p] : null, object);
 
+//
+// Returns a copy of the provided object with the provided value set
+// at path specified in keys
```

Comment:
This one is a little bit gnarly. Previous version was incorrectly mutating the `data` in place and causing all sorts of problematic side effects. This one generates a new object with the updated values to be passed back and set properly in react state.